packages/expo-av/android/src/main/java/expo/modules/av/progress/ProgressLooper.kt

```diff
@@ -0,0 +1,74 @@
+package expo.modules.av.progress
+
+typealias TimeMachineTick = () -> Unit
+
+interface TimeMachine {
+  fun scheduleAt(intervalMillis: Long, callback: TimeMachineTick)
+  val time: Long
+}
+
+/**
+ * @mczernek: NOTE that this class has corresponding test suite. However, since I was unable to set
+ * them up correctly, in order to run them, you should uncomment dependencies in module's build.gradle,
+ * and contents of test class: at packages/expo-av/android/src/test/java/expo/modules/av/progress/ProgressLooperTest.kt
+ * Note, that this breaks client's build, so remember to comment it back.q
+ */
+typealias PlayerProgressListener = () -> Unit
+
+class ProgressLooper(private val timeMachine: TimeMachine) {
+
+  private var interval = 0L
+  private var nextExpectedTick = -1L
+  private var waiting = false
+
+  private var shouldLoop: Boolean
+    get() = interval > 0 && nextExpectedTick >= 0 && !waiting
+    set(value) {
+      if (!value) {
+        interval = 0L
+        nextExpectedTick = -1L
+        waiting = false
+      }
+    }
+
+  private var listener: PlayerProgressListener? = null
+
+  fun changeListener(listener: PlayerProgressListener) {
```

Comment:
```suggestion
  fun setListener(listener: PlayerProgressListener) {
```
I believe this is more Android-style 🤔
[Here View example](https://developer.android.com/reference/android/view/View?hl=pt-br#setOnApplyWindowInsetsListener(android.view.View.OnApplyWindowInsetsListener )

Reply:
TBH, I don't know. What I wanted to emphasize here is the fact, that we actually change the listener, not set another one.
I know there is distinction between `set` and `add` which makes just that, so I'll go with that if current name confuses you.